Markdown Interface Updates

RC-Lee opened the pull request:
Worked on issue #9167 
 
  - "Editor" button text fixed to "Edit"
  - Edit and Preview buttons stay highlighted based on current mode
  - Added interface option for Editor and Preview Font type
  - Copied WYSIWYG CSS correct stylings to Markdown
  - Added custom block option placeholders for each input
  - Changed icon heading to use format_size icon
  - Added toolbar controls, toolbar also stays hidden in preview
 
![unknown](https://user-images.githubusercontent.com/32626950/139565640-dab7c13e-0280-4178-b67e-9c57aba113ef.png)

![unknown](https://user-images.githubusercontent.com/32626950/139565678-03040567-fcdf-4645-a924-49271271e4de.png)

![unknown](https://user-images.githubusercontent.com/32626950/139565706-206133e2-356f-4d05-8bbc-de0b539347a3.png)



Fixes #9167

RC-Lee committed:
Markdown Interface Updates
  - "Editor" button text fixed to "Edit"
  - Edit and Preview buttons stay highlighted based on current mode
  - Added interface option for Editor and Preview Font type
  - Copied WYSIWYG CSS correct stylings to Markdown
  - Added custom block option placeholders for each input
  - Changed icon heading to use format_size icon
  - Added toolbar controls

RC-Lee committed:
Markdown Interface Updates
  - "Editor" button text fixed to "Edit"
  - Edit and Preview buttons stay highlighted based on current mode
  - Added interface option for Editor and Preview Font type
  - Copied WYSIWYG CSS correct stylings to Markdown
  - Added custom block option placeholders for each input
  - Changed icon heading to use format_size icon
  - Added toolbar controls

azrikahar commented on app/src/interfaces/input-rich-text-md/index.ts:
```diff
@@ -43,6 +154,10 @@ export default defineInterface({
 								interface: 'input',
 								width: 'half',
+								options: {
+									placeholder: '$t:Name',
```
```suggestion
									placeholder: '$t:name',
```

azrikahar commented on app/src/interfaces/input-rich-text-md/input-rich-text-md.vue:
```diff
@@ -1,10 +1,15 @@
 		<div class="toolbar">
-			<v-menu show-arrow placement="bottom-start">
+			<v-menu
+				v-if="view[0] !== 'preview' && toolbar.indexOf('heading') !== -1"
```
We can probably use a `<template v-if="view[0] !== 'preview' ">` to wrap all these toolbar buttons (until before the spacer div) so we don't need to add `view[0] !== 'preview'` to every v-if 👍

As for `toolbar.indexOf('heading') !== -1`, technically it's true `indexOf` has better browser support, but maybe doing `toolbar.includes('heading')` is slightly more readable. Plus we don't support IE anymore 😉

azrikahar commented on app/src/interfaces/input-rich-text-md/input-rich-text-md.vue:
```diff
@@ -104,26 +131,34 @@
 			<v-item-group v-model="view" class="view" mandatory rounded>
-				<v-button x-small value="editor">Editor</v-button>
-				<v-button x-small value="preview">Preview</v-button>
+				<v-button x-small value="editor" :class="[{ active: view[0] !== 'preview' }]">Edit</v-button>
```
Since we're here, l think we can add translation strings to "Edit" and "Preview" as well, just for completeness sake in terms of our ongoing effort of whole app translations.

rclee91 replied:
Thanks for your feedback!
I've made a commit that should resolve all these requests.

azrikahar commented on app/src/interfaces/input-rich-text-md/input-rich-text-md.vue:
```diff
@@ -327,71 +389,85 @@ textarea {
+	color: var(--foreground-normal-alt);
+	font-weight: 700;
+	font-size: 36px;
+	font-family: v-bind(previewFont), serif;
```
TIL this is really interesting! Nice one

RC-Lee committed:
Refactoring based on review
  - Edit translation string for name in placeholder
  - Wrapped preview hidden toolbars with a <template v-if>
  - Changed .indexOf to .includes
  - Added translation strings for Edit and Preview

RC-Lee committed:
Refactoring based on review
  - Edit translation string for name in placeholder
  - Wrapped preview hidden toolbars with a <template v-if>
  - Changed .indexOf to .includes
  - Added translation strings for Edit and Preview

RC-Lee committed:
Refactoring - took off uncessary text

RC-Lee committed:
Refactoring - took off uncessary text

azrikahar commented:
@rclee91 Thanks for the prompt commit to resolve the requested changes!

Also appreciate the "drive-by" changes to use the CSS variables where applicable.

benhaynes commented:
Thanks @rclee91!! We'll review! ❤️

rijkvanzanten committed:
Merge branch 'main' into issue/9167

rijkvanzanten reviewed (changes requested):
It seems like the editor has some trouble loading in the fonts:

<img width="547" alt="CleanShot 2021-11-01 at 18 26 10@2x" src="https://user-images.githubusercontent.com/9141017/139750197-951fb22d-b042-4880-8734-76c0504e43c0.png">

rclee91 commented:
> 
> 
> It seems like the editor has some trouble loading in the fonts:
> 
I'm not sure why but it seems to work for me

![unknown](https://user-images.githubusercontent.com/32626950/139752064-340ac45d-f20d-4223-8138-c450b09b2d12.png)
![unknown](https://user-images.githubusercontent.com/32626950/139752090-155c6a78-4d16-4df9-adc4-22f594910da7.png)
![unknown](https://user-images.githubusercontent.com/32626950/139752153-d98f3c41-573a-4b0b-ba54-7f524dbcf352.png)

rijkvanzanten commented:
Ahh but those aren't the correct fonts though. We should make sure it's using the var(--family-sans-serif) et al. fonts to match the design language of the app 👍🏻

RC-Lee committed:
Changed to use --family- fonts

RC-Lee committed:
Merge branch 'directus:main' into issue/9167

RC-Lee committed:
merged with main

rclee91 commented:
> 
> 
> Ahh but those aren't the correct fonts though. We should make sure it's using the var(--family-sans-serif) et al. fonts to match the design language of the app 👍🏻

Newest commit [874783a](https://github.com/directus/directus/pull/9267/commits/8c4783a96e93539ec37dac406eec7866830f30f0) gives these styles now. 
I hope they are the correct ones.
![unknown](https://user-images.githubusercontent.com/32626950/139757232-a83e1876-c898-4d8b-898e-1dc7720f0279.png)
![unknown](https://user-images.githubusercontent.com/32626950/139757243-5a378657-4d2d-4861-95e1-4b1f5296e688.png)
![unknown](https://user-images.githubusercontent.com/32626950/139757251-d1ec1d11-b526-44d7-a8d5-8c15ac84a6db.png)

joselcvarela committed:
Merge remote-tracking branch 'origin/main' into issue/9167

joselcvarela commented on app/src/interfaces/input-rich-text-md/index.ts:
```diff
@@ -24,6 +24,117 @@ export default defineInterface({
+					choices: [
+						{
+							value: 'heading',
+							text: '$t:wysiwyg_options.heading',
```
Not sure about `t:wysiwyg_options...`
We could use `t:editors...` but we will lose all the translations.
So, for instance, I guess we will keep that way.

rijkvanzanten replied:
I think it's fine. We call the "WYSIWYG" interface the _input rich text html_ input, and this markdown one the _input rich text markdown_. Both are "what you see is what you get" style interfaces 👍🏻

joselcvarela commented on changelog.md:
```diff
@@ -7,162 +7,272 @@ _Changes marked with a :warning: contain potential breaking changes depending on
 ### :rocket: Improvements
 
 - **create-directus-project**
```
@rijkvanzanten I believe this changes were introduced by me, but I only merged main with this branch 😐

azrikahar replied:
I believe this is the same as last time we talked about, where the changelog/docs weren't linted, but merging it ended up linting it. No worries 👍

rijkvanzanten committed:
Merge branch 'main' into issue/9167
